plugins/inputs/marklogic/marklogic.go

```diff
@@ -0,0 +1,231 @@
+package marklogic
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Marklogic configuration toml
+type Marklogic struct {
+	URL      string   `toml:"url"`
+	Hosts    []string `toml:"hosts"`
+	Username string   `toml:"username"`
+	Password string   `toml:"password"`
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+type MlPointInt struct {
+	Value int `json:"value"`
+}
+
+type MlPointFloat struct {
+	Value float64 `json:"value"`
+}
+
+type MlPointBool struct {
+	Value bool `json:"value"`
+}
+
+// MarkLogic v2 management api endpoint for hosts status
+const statsPath = "/manage/v2/hosts/"
+const viewFormat = "?view=status&format=json"
+
+type MlHost struct {
+	HostStatus struct {
+		ID               string `json:"id"`
+		Name             string `json:"name"`
+		StatusProperties struct {
+			Online         MlPointBool `json:"online"`
+			LoadProperties struct {
+				TotalLoad MlPointFloat `json:"total-load"`
+			} `json:"load-properties"`
+			RateProperties struct {
+				TotalRate MlPointFloat `json:"total-rate"`
+			} `json:"rate-properties"`
+			StatusDetail struct {
+				Cpus                   MlPointInt `json:"cpus"`
+				Cores                  MlPointInt `json:"cores"`
+				TotalCPUStatUser       float64    `json:"total-cpu-stat-user"`
+				TotalCPUStatSystem     float64    `json:"total-cpu-stat-system"`
+				TotalCPUStatIdle       float64    `json:"total-cpu-stat-idle"`
+				TotalCPUStatIowait     float64    `json:"total-cpu-stat-iowait"`
+				MemoryProcessSize      MlPointInt `json:"memory-process-size"`
+				MemoryProcessRss       MlPointInt `json:"memory-process-rss"`
+				MemorySystemTotal      MlPointInt `json:"memory-system-total"`
+				MemorySystemFree       MlPointInt `json:"memory-system-free"`
+				MemorySize             MlPointInt `json:"memory-size"`
+				HostSize               MlPointInt `json:"host-size"`
+				DataDirSpace           MlPointInt `json:"data-dir-space"`
+				QueryReadBytes         MlPointInt `json:"query-read-bytes"`
+				QueryReadLoad          MlPointInt `json:"query-read-load"`
+				HTTPServerReceiveBytes MlPointInt `json:"http-server-receive-bytes"`
+				HTTPServerSendBytes    MlPointInt `json:"http-server-send-bytes"`
+			} `json:"status-detail"`
+		} `json:"status-properties"`
+	} `json:"host-status"`
+}
+
+// Description of plugin returned
+func (c *Marklogic) Description() string {
+	return "Retrives information on a specific host in a MarkLogic Cluster"
+}
+
+var sampleConfig = `
+  ## Base URL of the MarkLogic HTTP Server.
+  url = "http://localhost:8002"
+
+  ## List of specific hostnames to retrieve information. At least (1) required.
+  # hosts = ["hostname1", "hostname2"]
+
+  ## Using HTTP Digest Authentication. Management API requires 'manage-user' role privileges
```

Comment:
Update the comment for HTTP Basic Auth, don't forget to also update it in the README.